src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -670,39 +673,43 @@ CHIP_ERROR AdvertiserMinMdns::Advertise(const CommissionAdvertisingParameters &
         }
     }
 
-    if (params.GetVendorId().HasValue())
     {
-        MakeServiceSubtype(nameBuffer, sizeof(nameBuffer),
-                           DiscoveryFilter(DiscoveryFilterType::kVendorId, params.GetVendorId().Value()));
-        FullQName vendorServiceName =
-            allocator->AllocateQName(nameBuffer, kSubtypeServiceNamePart, serviceType, kCommissionProtocol, kLocalDomain);
-        ReturnErrorCodeIf(vendorServiceName.nameCount == 0, CHIP_ERROR_NO_MEMORY);
-
-        if (!allocator->AddResponder<PtrResponder>(vendorServiceName, instanceName)
-                 .SetReportAdditional(instanceName)
-                 .SetReportInServiceListing(true)
-                 .IsValid())
+        const auto & vendorId = params.GetVendorId();
+        if (vendorId.has_value())
```

Comment:
Super nit but in C++17, you can actually get rid off those scopes with:
```suggestion
        if (const auto & vendorId = params.GetVendorId(); vendorId.has_value())
```

Reply:
This is neat! I will do so.